integrationTests/chainSimulator/bridge/executeOrDepositFail_test.go

```diff
@@ -0,0 +1,370 @@
+	bridgedOutTokens []chainSim.ArgsDepositToken,
+) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+
+	roundsPerEpoch := core.OptionalUint64{
+		HasValue: true,
+		Value:    20,
+	}
+
+	whiteListedAddress := "erd1qqqqqqqqqqqqqpgqcw92wj0huvaghg4aeuykknp7hstmrmhudz3shjdhtt"
+	cs, err := chainSimulator.NewChainSimulator(chainSimulator.ArgsChainSimulator{
+		BypassTxSignatureCheck:   true,
+		TempDir:                  t.TempDir(),
+		PathToInitialConfig:      defaultPathToInitialConfig,
+		NumOfShards:              1,
+		GenesisTimestamp:         time.Now().Unix(),
+		RoundDurationInMillis:    uint64(6000),
+		RoundsPerEpoch:           roundsPerEpoch,
+		ApiInterface:             api.NewNoApiInterface(),
+		MinNodesPerShard:         3,
+		MetaChainMinNodes:        3,
+		NumNodesWaitingListMeta:  0,
+		NumNodesWaitingListShard: 0,
+		AlterConfigsFunction: func(cfg *config.Configs) {
+			cfg.GeneralConfig.VirtualMachine.Execution.TransferAndExecuteByUserAddresses = []string{whiteListedAddress}
+		},
+	})
+	require.Nil(t, err)
+	require.NotNil(t, cs)
+
+	defer cs.Close()
+
+	err = cs.GenerateBlocksUntilEpochIsReached(3)
+	require.Nil(t, err)
+
+	nodeHandler := cs.GetNodeHandler(0)
+
+	// Deploy bridge setup
+	initialAddress := "erd1l6xt0rqlyzw56a3k8xwwshq2dcjwy3q9cppucvqsmdyw8r98dz3sae0kxl"
+	argsEsdtSafe := ArgsEsdtSafe{
+		ChainPrefix:       prefix,
+		IssuePaymentToken: "ABC-123456",
+	}
+	bridgeData := deployBridgeSetup(t, cs, initialAddress, simpleEsdtSafeWasmPath, argsEsdtSafe, feeMarketWasmPath)
+
+	// esdt-safe address generated is NOT whitelisted
+	esdtSafeEncoded, _ := nodeHandler.GetCoreComponents().AddressPubKeyConverter().Encode(bridgeData.ESDTSafeAddress)
+	require.NotEqual(t, whiteListedAddress, esdtSafeEncoded)
+
+	walletAddress := "erd1j7t00tq4jhcarcl2kuw3aywplnqlh9y6w9sxx2v6namc3jxvn5qq0jumrh"
+	err = cs.SetStateMultiple([]*dtos.AddressState{
+		{
+			Address: walletAddress,
+			Balance: "10000000000000000000000",
+		},
+	})
+	require.Nil(t, err)
+	walletAddrBytes, _ := nodeHandler.GetCoreComponents().AddressPubKeyConverter().Decode(walletAddress)
+
+	wallet := dtos.WalletAddress{Bech32: walletAddress, Bytes: walletAddrBytes}
+	nonce := uint64(0)
+
+	for _, token := range bridgedOutTokens {
+		tokenData := esdt.ESDigitalToken{
+			Value:         token.Amount,
+			Type:          uint32(token.Type),
+			TokenMetaData: getTokenMetaData(token),
+		}
+		chainSim.GetEsdtInWallet(t, cs, wallet, token.Identifier, token.Nonce, tokenData)
+
+		err = cs.GenerateBlocks(1)
+		require.Nil(t, err)
+		chainSim.RequireAccountHasToken(t, cs, getTokenIdentifier(token), wallet.Bech32, token.Amount)
+	}
+
+	// deposit an array of tokens from main chain to sovereign chain,
+	// expecting these tokens to NOT be burned by ESDT safe sc because is not whitelisted
+	txResult := deposit(t, cs, wallet.Bytes, &nonce, bridgeData.ESDTSafeAddress, bridgedOutTokens, wallet.Bytes)
+	chainSim.RequireSignalError(t, txResult, actionNotAllowed)
```

Comment:
Same question as above

Reply:
The error is `signalError` event with `action is not allowed` and is checked. This is what VM is returning. Do we expect something else?